### ISSUE TITLE
/tool tree command producing trees with inadequate amount of log blocks

### ISSUE THREAD

**Issue by yngmnyaustin** [EngineHub/WorldEdit#2508]
### WorldEdit Version

7.3.0

### Platform Version

Forge 1.20.4-49.0.34

### Confirmations

- [X] I am using the most recent Minecraft release.
- [X] I am using a version of WorldEdit compatible with my Minecraft version.
- [X] I am using the latest or recommended version of my platform software.
- [X] I am NOT using a hybrid server, e.g. a server that combines Bukkit and Forge. Examples include Arclight, Mohist, and Cardboard.
- [X] I am NOT using a fork of WorldEdit, such as FastAsyncWorldEdit (FAWE) or AsyncWorldEdit (AWE)

### Bug Description

When using the `/tool tree [type]` command to generate trees (i.e. [type] = spruce, bigtree, etc.) the resulting trees generated by the command contain an inadequate amount of log blocks, causing leaf blocks to decay unless additional log blocks are manually placed.

### Expected Behavior

When using the `/tool tree [type]` command, the generated tree should contain a number of log blocks, evenly spaced within the leaf blocks, in order to prevent automatic decay of the generated leaf blocks.

### Reproduction Steps

1. Bound the `/tool tree [type]` command to an eligible item (i.e. Stick)
2. Choose a block to generate the tree on (i.e. Grass)
3. Right click to generate tree. The generated tree will not contain enough log blocks to prevent leaf decay.

**Comment by yngmnyaustin** [comment]
Ah, I see. I thought this was a bug as I've been able to use the command to generate trees that didn't decay in previous versions. Perhaps the generation code has changed in a new update? Good to know though.

**Comment by Azutte** [comment]
![image](https://github.com/user-attachments/assets/7887b49c-2207-4fe3-972f-3d06f62cae5b)
![2024-07-20_11 33 35](https://github.com/user-attachments/assets/30d98eeb-f00c-4cfe-907c-6cf927866057)
I look at these and can identify spots where it's 100% supposed to be logs (green lines in the first one are my addition), in the second image it even has zero trunk. This happens with tool set to `tree largeoak` and it also happens with `//feature fancy_oak`.

If I set up a bunch of trees with shafts of cobblestone above them (known way to force large oaks in survival) and just start autoclicking bonemeal onto them, this happens [zero percent of the time](https://imgur.com/a/cvw4q9b), leading me to believe it's not the intended vanilla experience. One of those is even the size of the trunkless image from earlier, but has the trunk this time.

Looking around I spotted [this](https://gist.github.com/Earthcomputer/41addf80c12d001dfa4391c3a0d03be8) explanation of large/big/fancy oak generation which claims the following:

> `leafDistanceLimit`: The distance from a log that leaves can generate, set to 4 for sapling growth, though appears to be set to 5 for natural growth for some reason

I could see this being the root of the problem I'm seeing -- if the worldgen tree feature generator still works like this, then worldedit calling it is just happening to give me branches with lengths short enough where their leaves block spheres overlap where the trunk goes. But I don't know where to look to see if that's how tree generation still works beyond the fact that I can't see it happening in several dozen bonemealed examples.

Everything tested in fabric.
